src/components/calendar/calendar.spec.js

```diff
@@ -38,13 +40,106 @@ describe('md-checkbox', function() {
       var selectedDate = element.querySelector('.md-calendar-selected-date');
 
       expect(displayedMonth.textContent).toBe('May');
-      expect(selectedDate.textContent).toBe('30')
+      expect(selectedDate.textContent).toBe('30');
     });
 
   });
 
   describe('calendar construction', function() {
 
+    /**
+     * Extracts an text as an array from a tr element.
+     */
+    function extractRowText(tr) {
+      var headerTexts = [];
+      angular.forEach(tr.children, function(tableElement) {
+        headerTexts.push(tableElement.textContent);
+      });
+
+      return headerTexts;
+    }
+
+    /**
+     * Finds a date td given a day of the month from an .md-calendar-month element.
+     */
+    function findDateElement(monthElement, day) {
+      var tds = monthElement.querySelectorAll('td');
+      var td;
+
+      for (var i = 0; i < tds.length; i++) {
+        td = tds[i];
+        if (td.textContent === day.toString()) {
+          return td;
+        }
+      }
+    }
```

Comment:
Can you move these helper functions to the top of the file (under `applyDateChange`) ?